src/pyatsimagebuilder/builder.py

```diff
@@ -74,33 +72,32 @@ def run(self, keep_context=False, tag=None,no_cache=True,dry_run=False):
         self.context = Context(keep=keep_context, logger=self._logger)
 
         with self.context:
-            
-            # create our installation directory 
+
+            # create our installation directory
             self.context.mkdir(INSTALLATION)
-            self.context.mkdir(INSTALLATION/REQUIREMENTS)
+            self.context.mkdir(INSTALLATION / REQUIREMENTS)
 
             self._populate_context()
 
             # Tag for docker image   argument (cli) > config (yaml) > None
-            self.image.tag = tag or self.config.get('tag', None)  
+            self.image.tag = tag or self.config.get('tag', None)
 
             # Start docker build
             if not dry_run:
                 self._logger.info('Building image')
                 self._build_image(no_cache=no_cache)
-                self._logger.info("Built image '%s' successfully" 
-                                    % tag if tag else self.image.id)
-
+                self._logger.info("Built image '%s' successfully" %
+                                  tag if tag else self.image.id)
 
         return self.image
-        
+
     def _populate_context(self):
 
         # Dump config to yaml file in context
         self._logger.info('Saving config to context folder')
-        self.context.write_file(INSTALLATION / 'build.yaml',
-                                yaml.safe_dump(self.config, 
-                                               default_flow_style=False))
+        self.context.write_file(
+            INSTALLATION / 'build.yaml',
+            yaml.safe_dump(self.config, default_flow_style=False))
 
```

Comment:
The yaml file written to the image will include any credentials/ssh keys. If the image is uploaded to a repository or shared with another user, that information will be exposed. You might want to move this to after the dockerfile writing, so that sensitive information can be removed first.

Reply:
Move this to the end of `_populate_context` and pop the credentials/ssh_key from the `self.config` so they are not written in